src/gui/src/layoutViewer.cpp

```diff
@@ -1973,6 +1976,29 @@ void LayoutViewer::drawBlock(QPainter* painter,
   }
 }
 
+void LayoutViewer::drawAccessPoints(Painter& painter)
+{
+  for (auto term : block_->getITerms()) {
+    for (auto ap : term->getAccessPoints()) {
+      if (ap != nullptr && options_->isVisible(ap->getLayer())) {
```

Comment:
The returned vector of getAccessPoints is a vector of the preferred access point for each pin on the term ordered. some pins may not have an access point, I store it as nullptr. I have been thinking about this and I am not really sure if it's ok, but I am following TritonRoute's API.

Reply:
I think it makes more sense to store only the access points disregarding order to pins. the user of the api can infer the pins from the access points. If you agree, I will remove adding the null pointers.

Reply:
@maliberty need your opinion to resolve this.

Reply:
I think my other comments about using a map rather than a vector would resolve this question.   You would only put in the map what is actually populated.